### PR TITLE
fixed a bug where mptable failed to wake up CPU1# when adding acpi=off into User VM cmdline

### PR DESCRIPTION
**Pull request by donsheng**
Previous commit introduced the "use physical APIC IDs as vLAPIC IDs for VMs" change,
but it didn't update the apic_id in mptable. Changed the mptable code to also
set apic_id to physical APIC ID to fix the bug

Tracked-On: #7146

### CODE REVIEW

**Comment by acrnsi** [comment]
Can one of the admins verify this patch?

**Comment by wenlingz** [comment]
@ywan170 please help approve this PR, thanks

File: devicemodel/core/mptbl.c [hunk 1]
```diff
@@ -117,11 +118,39 @@ static void
...
+		pr_err("%s,Err: Invalid guest_pcpu_bitmask.\n", __func__);
+		return;
+	}
+
+	pr_info("%s, guest_cpu_bitmask: 0x%x\n", __func__, guest_pcpu_bitmask);
 
 	for (i = 0; i < ncpu; i++) {
+		int pcpu_id = pcpuid_from_vcpuid(guest_pcpu_bitmask, i);
+		int lapic_id;
+
```
Comment (lifeix) [inline devicemodel/core/mptbl.c:133]: Do we need to check whether the pcpu_id is valid ? Now the lapicid_from_pcpuid will also does this check.

Comment (donsheng) [reply devicemodel/core/mptbl.c:133]: Hi,lifei,
yes, this patch does including the pcpu_id validity checking, see below code in this patch:
if (pcpu_id < 0) {
			pr_err("%s,Err: pcpu id is not found in guest_pcpu_bitmask.\n", __func__);
			return;
		}

		assert(pcpu_id < ACRN_PLATFORM_LAPIC_IDS_MAX);
		if (pcpu_id >= ACRN_PLATFORM_LAPIC_IDS_MAX) {
			pr_err("%s,Err: pcpu id %u should be less than ACRN_PLATFORM_LAPIC_IDS_MAX.\n", __func__, pcpu_id);
			return;
		}

**Comment by wenlingz** [comment]
@ywan170 please help approve this PR, thanks

File: devicemodel/core/mptbl.c (hunk 1 above)
Comment (donsheng) [reply devicemodel/core/mptbl.c:133]: @lifeix  can you double check my patch? it is already checking pcpu_id

**Review by ywan170 — APPROVED** [review]

**Pull request merged**